hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieLogFormatReader.java

```diff
@@ -104,8 +90,6 @@ public boolean hasNext() {
         // First close previous reader only if readBlockLazily is false
         if (!readBlocksLazily) {
```

Comment:
I don't understand, how could we eliminate the memory usage if we do not close the reader eagerly, the input streams are still there?

Reply:
my bad. missed the fix here after raising a new patch. 
I should call this.currentReader.close() here irrespective of whether its lazy or not.